packages/react-components/src/components/utils/Datetime.ts

```diff
@@ -75,9 +75,31 @@ export const formatTimestampForChatMessage = (messageDate: Date, todayDate: Date
   // If message was before first day of the week then timestamp string is like Monday 1:30 p.m.
   const firstDayOfTheWeekDate = new Date(todayDate.getFullYear(), todayDate.getMonth(), todayDate.getDate() - weekDay);
   if (messageDate > firstDayOfTheWeekDate) {
-    return weekDayToDayName[messageDate.getDay()] + ' ' + formatTimeForChatMessage(messageDate);
+    const day = weekDayToDayName(messageDate.getDay(), strings);
+    return (day !== '' ? day : formatDateForChatMessage(messageDate)) + ' ' + formatTimeForChatMessage(messageDate);
   }
 
   // If message date is in previous or older weeks then timestamp string is like 2021-01-10 1:30 p.m.
   return formatDateForChatMessage(messageDate) + ' ' + formatTimeForChatMessage(messageDate);
 };
+
+const weekDayToDayName = (day: number, strings: ILocaleKeys): string => {
```

Comment:
nit: just `dayToDayName`, weekDay looks like a typo of weekday implying just Mon-Fri